File: striver/S1L4/java/Palindrome.java
import java.util.Scanner;

public class Palindrome {
    public static boolean isPalindrome(int num) {
        int original = num;
        int reversedNum = 0;
        while (num > 0) {
            int digit = num % 10;
            reversedNum = reversedNum * 10 + digit;
            num /= 10;
        }
        return original == reversedNum;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int num = sc.nextInt();
        if (isPalindrome(num)) {
            System.out.println(num + " is a palindrome");
        } else {
            System.out.println(num + " is not a palindrome");
        }
        sc.close();
    }
} 

File: striver/S1L4/python/palindrome.py
def is_palindrome(num):
    """
    Check if a number is palindrome
    Example: 121 -> True, 123 -> False
    """
    original = num
    reversed_num = 0
    while num > 0:
        digit = num % 10
        reversed_num = reversed_num * 10 + digit
        num //= 10
    return original == reversed_num

if __name__ == "__main__":
    num = int(input("Enter a number: "))
    if is_palindrome(num):
        print(f"{num} is a palindrome")
    else:
        print(f"{num} is not a palindrome") 